tests/pipelines/test_pipelines_common.py

```diff
@@ -731,7 +734,9 @@ def _test_xformers_attention_forwardGenerator_pass(
         )
 
         if test_max_difference:
-            max_diff = np.abs(output_with_offload - output_without_offload).max()
+            max_diff = numpy_cosine_similarity_distance(
+                to_np(output_with_offload).flatten(), to_np(output_without_offload).flatten()
```

Comment:
https://github.com/huggingface/diffusers/pull/4812/files#r1307509747 would be better